src/emqx_auth_username.erl

```diff
@@ -135,12 +137,13 @@ description() ->
 %% Internal functions
 %%-----------------------------------------------------------------------------
 
-hash(Password) ->
-    SaltBin = salt(), <<SaltBin/binary, (md5_hash(SaltBin, Password))/binary>>.
+encrypted_data(Password) ->
+    HashType = application:get_env(emqx_auth_username, password_hash, md5),
+    SaltBin = salt(),
+    <<SaltBin/binary, (hash(Password, SaltBin, HashType))/binary>>.
 
-md5_hash(SaltBin, Password) ->
-    erlang:md5(<<SaltBin/binary, Password/binary>>).
+hash(Password, SaltBin, HashType) ->
+    emqx_passwd:hash(HashType, <<SaltBin/binary, Password/binary>>).
 
 salt() ->
-    emqx_time:seed(), Salt = rand:uniform(16#ffffffff), <<Salt:32>>.
-
+    emqx_time:seed(), Salt = rand:uniform(16#ffffffff), <<Salt:32>>.
```

Comment:
It looks like the result of expression `emqx_time:seed` is not used.
I think writing `rand:seed(exsplus, erlang:timestamp())`directly is a better way and easy to understand.

Reply:
yeah! I remove it